### PR TITLE
Cleaned up ISO-TP implementation

### PR DESCRIPTION
**Pull request by akorb**
Usual cleaning work.

The ISO-TP class overrode several methods and implemented them in the same way.

### CODE REVIEW

File: scapy/contrib/isotp.py [hunk 1]
```diff
@@ -328,18 +319,7 @@ def recv_raw(self, x=0xffff):
...
 
-    def send(self, x):
-        if hasattr(x, "sent_time"):
-            x.sent_time = time.time()
-        return self.outs.send(bytes(x))
-
-    def sr(self, *args, **kargs):
-        return sndrcv(self, *args, **kargs)
-
     def sr1(self, *args, **kargs):
```
Comment (gpotter2) [inline scapy/contrib/isotp.py]: I think you can also remove `sr1`

See https://github.com/secdev/scapy/blob/master/scapy/supersocket.py#L88

Comment (akorb) [reply scapy/contrib/isotp.py]: I understand. The first `[0]` is for the tuple access.

I agree with you. Is this fine for you, @polybassa ?

Comment (polybassa) [reply scapy/contrib/isotp.py]: can you please check, if these unit-test still work, after your changes. Unfortunately these test run only if the kernel module is installed, therefore they are not part of the continuous integration server.

`# isotp unit tests
#
# Type the following command to launch start the tests:
# $ sudo bash test/run_tests -t test/isotp.uts -F

% isotp unit tests
~ python3_only

+ Configuration of scapy

= Setup string for vcan
~ conf command
bashCommand1 = "/bin/bash -c 'sudo modprobe can; sudo modprobe vcan; sudo ip link add name vcan0 type vcan; sudo ip link set dev vcan0 up'"

= Load os
~ conf command needs_root linux
import os
import threading
from time import sleep
import subprocess

p = subprocess.Popen(['bash', '-c', 'lsmod | grep "^can_isotp"'], stdout=subprocess.PIPE)
sleep(0.2)
HAS_ISOTP_INSTALLED = b"can_isotp" in p.stdout.readline()

= Load isotp socket and setup vcan
~ conf command needs_root linux
if HAS_ISOTP_INSTALLED:
    0 == os.system(bashCommand1)

if HAS_ISOTP_INSTALLED:
    conf.contribs['CANSocket'] = {'use-python-can': True}
    load_layer("can")
    from scapy.contrib.cansocket_python_can import *
    import can
    load_contrib("isotp")


+ Basic Socket Tests()

= ISOTP Socket simple send
~ needs_root linux
if HAS_ISOTP_INSTALLED:
    def sender():
        sleep(0.2)
        sock = ISOTPSocket('vcan0', 123, 321)
        sock.send(ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33'))
    
    
    txThread = threading.Thread(target=sender)
    sock = ISOTPSocket('vcan0', 321, 123)
    txThread.start()
    rx = sock.recv()
    rx == ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
    rx.show()
    bytes(rx)
    bytes(ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33'))
    txThread.join()

= ISOTP Socket send with extended addressing
~ needs_root linux
if HAS_ISOTP_INSTALLED:
    def sender():
        sleep(0.2)
        sock = ISOTPSocket('vcan0', 123, 321, 23, 21)
        sock.send(ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33'))
    
    txThread = threading.Thread(target=sender)
    sock = ISOTPSocket('vcan0', 321, 123, 21, 23)
    txThread.start()
    rx = sock.recv()
    rx == ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
    txThread.join()

= ISOTP Socket validate single frame
~ needs_root linux
if HAS_ISOTP_INSTALLED:
    def sender():
        sleep(0.1)
        sock = ISOTPSocket('vcan0', 0x123, 0x321)
        sock.send(Raw(load=b'\x11\x22\x33'))
    
    txThread = threading.Thread(target=sender)
    sock = CANSocket(iface=can.interface.Bus(bustype='socketcan', channel='vcan0', bitrate=250000))
    txThread.start()
    rx = sock.recv()
    tx = CAN(identifier=0x321, length=4, data=b'\x03\x11\x22\x33')
    rx.data == tx.data
    rx.length == tx.length
    rx.identifier == tx.identifier
    txThread.join()
    
= ISOTP Socket sr1 test
~ needs_root linux
if HAS_ISOTP_INSTALLED:
    def sender():
        sleep(0.1)
        sock = ISOTPSocket('vcan0', 0x123, 0x321)
        rx = sock.sr1(Raw(load=b'\x11\x22\x33'), timeout=1, verbose=True)
        assert(rx is not None)
        assert(rx == ISOTP(b'\x7f\x22\x33'))
    
    txThread = threading.Thread(target=sender)
    sock = CANSocket(iface=can.interface.Bus(bustype='socketcan', channel='vcan0', bitrate=250000))
    txThread.start()
    rx = sock.recv()
    rx == CAN(identifier=0x321, length=4, data=b'\x03\x11\x22\x33')
    sock.send(CAN(identifier=0x123, length=4, data=b'\x03\x7f\x22\x33'))
    txThread.join()
    
= ISOTP Socket sr1 and ISOTP test
~ needs_root linux
if HAS_ISOTP_INSTALLED:
    def sender():
        sleep(0.1)
        sock = ISOTPSocket('vcan0', 0x123, 0x321)
        rx = sock.sr1(ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33'), timeout=1, verbose=True)
        rx is not None
        rx == ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
    
    txThread = threading.Thread(target=sender)
    sock = ISOTPSocket('vcan0', 0x321, 0x123)
    txThread.start()
    rx = sock.recv()
    rx == ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
    sock.send(ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33'))
    txThread.join()
    
= ISOTP Socket sr1 and ISOTP test vice versa
~ needs_root linux
if HAS_ISOTP_INSTALLED:
    def receiver():
        sock = ISOTPSocket('vcan0', 0x321, 0x123)
        rx = sock.recv()
        rx == ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
        sock.send(ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33'))
    
    rxThread = threading.Thread(target=receiver)
    rxThread.start()
    sleep(0.1)
    sock = ISOTPSocket('vcan0', 0x123, 0x321)
    rx = sock.sr1(ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33'), timeout=1,verbose=True)
    rx == ISOTP(b'\x11\x22\x33\x11\x22\x33\x11\x22\x33\x11\x22\x33')
    rxThread.join()
`

Comment (akorb) [reply scapy/contrib/isotp.py]:
```### Booting scapy...
### Starting tests...
### Loading: test/isotp.uts
passed EEEA08D4 Setup string for vcan
passed F83D3B46 Load os
RTNETLINK answers: File exists
passed 967B6976 Load isotp socket and setup vcan
passed 3A74C80A ISOTP Socket simple send
passed 8668A25F ISOTP Socket send with extended addressing
passed 11760C21 ISOTP Socket validate single frame
passed 6E3093CF ISOTP Socket sr1 test
passed 1492CA36 ISOTP Socket sr1 and ISOTP test
passed 9492E26B ISOTP Socket sr1 and ISOTP test vice versa
Campaign CRC=111BD9A0  SHA=64DBAA58AA25217861164A5CD314CA47EA634DBD
PASSED=9 FAILED=0
### Writing output...
isotp unit tests
-- Run Tue Sep  4 10:52:47 2018 from [test/isotp.uts] by UTscapy

Passed=9
Failed=0```

@polybassa

**Comment by polybassa** [comment]
I’m fine with it

**Pull request closed without merging**